src/basicio.cpp

```diff
@@ -1093,10 +1093,12 @@ namespace Exiv2 {
     void MemIo::Impl::reserve(long wcount)
     {
         long need = wcount + idx_;
+        long    blockSize =     32*1024;   // 32768           `
+        long maxBlockSize = 4*1024*1024;
 
         if (!isMalloced_) {
-            // Minimum size for 1st block is 32kB
-            long size  = EXV_MAX(32768 * (1 + need / 32768), size_);
+            // Minimum size for 1st block
+            long size  = EXV_MAX(blockSize * (1 + need / blockSize), size_);
             byte* data = (byte*)std::malloc(size);
```

Comment:
`malloc` can return `NULL` when the allocation request failed, we should check that here. Otherwise we'll get a segfault in the next line.